presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -651,7 +652,7 @@ std::unique_ptr<common::Filter> toFilter(
       return combineIntegerRanges(bigintFilters, nullAllowed);
     }
 
-    if (type->kind() == TypeKind::DATE) {
+    if (type->isDate()) {
```

Comment:
This block should be moved at line 642 before above conditional statement for TypeKind::INTEGER (see https://github.com/prestodb/presto/pull/19935/files#diff-b596594147f956ff5aadb512a14f1538d5975c551dad6ba773ac266762187734). 

Without that change the Velox PR will start using the above conditional block for TypeKind::INTEGER since it will map date to TypeKind::INTEGER and that will take precedence. Instead the date related code should take precedence.